src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -439,13 +439,22 @@ private void SetWorkingSetLimitsCore(IntPtr? newMin, IntPtr? newMax, out IntPtr
         /// <param name="startInfo">The start info with which to start the process.</param>
         private bool StartWithCreateProcess(ProcessStartInfo startInfo)
         {
+            string arguments;
             // See knowledge base article Q190351 for an explanation of the following code.  Noteworthy tricky points:
             //    * The handles are duplicated as non-inheritable before they are passed to CreateProcess so
             //      that the child process can not close them
             //    * CreateProcess allows you to redirect all or none of the standard IO handles, so we use
             //      GetStdHandle for the handles that are not being redirected
 
-            StringBuilder commandLine = BuildCommandLine(startInfo.FileName, startInfo.Arguments);
+            if (startInfo != null && startInfo.ArgumentList.Count > 0)
```

Comment:
You don't need to null-check `startInfo` - you're already surrounded by code that assumes a non-null `startInfo` as an invariant.